File: app/src/main/java/io/vinter/trackmyanime/ui/register/RegisterActivity.java
package io.vinter.trackmyanime.ui.register;

import android.arch.lifecycle.ViewModelProviders;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.EditText;
import android.widget.Toast;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import io.vinter.trackmyanime.R;
import io.vinter.trackmyanime.network.form.LoginForm;
import io.vinter.trackmyanime.ui.login.LoginViewModel;

public class RegisterActivity extends AppCompatActivity {

    @OnClick(R.id.register)
    void register(){
        if (!loginField.getText().toString().equals("") & !passwordField.getText().toString().equals("")){
            if (passwordField.getText().toString().equals(passwordConfirmField.getText().toString())){
                viewModel.register(new LoginForm(loginField.getText().toString(), passwordField.getText().toString()));
            } else {
                Toast.makeText(this, "Password does not match confirm", Toast.LENGTH_SHORT).show();
            }
        } else {
            Toast.makeText(this, "All fields are required", Toast.LENGTH_SHORT).show();
        }
    }

    @BindView(R.id.register_login)
    EditText loginField;

    @BindView(R.id.register_password)
    EditText passwordField;

    @BindView(R.id.register_confirm_password)
    EditText passwordConfirmField;

    RegisterViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);
        viewModel = ViewModelProviders.of(this).get(RegisterViewModel.class);
        ButterKnife.bind(this);

        viewModel.registration.observe(this, message -> {
            Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
            Intent intent = new Intent();
            setResult(RESULT_OK, intent);
            finish();
        });
    }
}

File: app/src/main/java/io/vinter/trackmyanime/ui/register/RegisterActivity.kt
package io.vinter.trackmyanime.ui.register

import android.app.Activity
import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast

import butterknife.ButterKnife
import io.vinter.trackmyanime.R
import io.vinter.trackmyanime.network.form.LoginForm
import kotlinx.android.synthetic.main.activity_register.*

class RegisterActivity : AppCompatActivity() {

    private lateinit var viewModel: RegisterViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)
        viewModel = ViewModelProviders.of(this).get(RegisterViewModel::class.java)
        ButterKnife.bind(this)

        val loginField = register_login
        val passwordField = register_password
        val passwordConfirmField = register_confirm_password
        register.setOnClickListener {
            if ((loginField.text.toString() != "") and (passwordField.text.toString() != "")) {
                if (passwordField.text.toString() == passwordConfirmField.text.toString()) {
                    viewModel.register(LoginForm(loginField.text.toString(), passwordField.text.toString()))
                } else {
                    Toast.makeText(this, "Password does not match confirm", Toast.LENGTH_SHORT).show()
                }
            } else {
                Toast.makeText(this, "All fields are required", Toast.LENGTH_SHORT).show()
            }
        }

        viewModel.registration.observe(this, Observer{
            Toast.makeText(this, it, Toast.LENGTH_SHORT).show()
            val intent = Intent()
            setResult(Activity.RESULT_OK, intent)
            finish()
        })

        viewModel.error.observe(this, Observer {
            if (it != null) {
                Toast.makeText(this, it, Toast.LENGTH_SHORT).show()
                viewModel.error.postValue(null)
            }
        })
    }
}
